specification/redis/resource-manager/Microsoft.Cache/preview/2023-05-01/examples/RedisCacheAccessPolicyAssignmentList.json

```diff
@@ -0,0 +1,44 @@
+{
+  "parameters": {
+    "subscriptionId": "subid",
+    "resourceGroupName": "rg1",
+    "cacheName": "cache1",
+    "api-version": "2023-05-01"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
+          {
+            "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Cache/redis/cache1/accessPolicyAssignments/accessPolicy1",
+            "name": "cache1/accessPolicy1",
```

Comment:
Oh I see, it is actually aliasing to the accessPolicy1 resource...? This is a kind of confusing difference to other ARM APIs.

Reply:
We're listing all the accessPolicyAssignments for accessPolicy1, thus using accessPolicy1 here

Reply:
Over here we're listing the assignments done for accessPolicy1, thus the naming

Reply:
Right but the URL for doing that is a bit surprising. Compared to e.g.
```
            ""/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Cache/redis/cache1/accessPolicies/accessPolicy1/assignments",
```
or
```
            "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Cache/redis/cache1/accessPolicyAssignments?$filter='accessPolicy eq accessPolicy1'"
```

Reply:
I agree with Tim that this is very unorthodox. Resources aren't referenced by name, they should be referenced by ID.  In this case its even more unusual in that the reference is implied by the name of the "assignment" that is being created being equal to the name of the policy that is being assigned.

I think this is not only unorthodox but problematic in your scenario because:

1. It forces all objectId assignments for a specific policy to exist as an array inside 1 resource. If there are a lot that resource is going to be large. That resource is also not going to be safe to update concurrently because PATCH doesn't work well on arrays (if it was implemented in this API) and peoples changes will stomp on one another.
2. It eliminates the possibility of doing linked access checks on the access policy you are assigning (since there is no resourceid in the body to do the access check on). 
3. RBAC becomes much less granular. An "owner" access policy has 1 and only 1 policy assignment associated with it and anyone that has access to it has access to add/remove any objectId from that assignment.  If they were individual assignments that referenced the policy in their body I could give someone access to a specific instance of the owner assignment eliminating the potential for them messing up the other owner assignments.

Reply:
@pilor , @TimLovellSmith , Thanks for pointing this out, we had initial offline discussion for the assignment api earlier with @TimLovellSmith and after that we moved ahead with this approach. We are open to changing our api to incorporate this suggestion in our stable GA release of the api version, but this is too late for this preview version of API as we have public preview release next week. Could you please help in merging this preview version with the exception? We will update our API version before we publish GA/stable version of the corresponding api version. Hope that will be ok.

Reply:
You won't be able to make breaking changes like this between public preview and stable API versions

Reply:
I see, then we will be able to create a completely new api-version for GA release, right? Like 2023-06-01?

Reply:
New API versions can still not have breaking changes.  New API versions are for expanding functionality.  The policy is here:  https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/overview_breakingchanges